### PR TITLE
DNF Sort Explanation added

### PR DESCRIPTION
**Pull request by Pranshu321**
## Related Issue
- #1173 

## Proposed Changes
- Detail Explanation of DNF sort in readme added and code for DNF sort is already added by someone but the explanation is missing so I added that to enhance the experience.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

In this I have added a detailed explanation of DNF sort starting from what it is and ending with the disadvantages, all document was written in a simple language to be understandable easily.

DNF Sort is the sorting method that is used to sort the array only having `0's` `1's` and `2's`.

Change is required to enhance the understanding of the DNF sort.

#1173 

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by Mrjoy832** [comment]
LGTM🌟

File: S-SortingAlgorithms/README.md [hunk 1]
```diff
@@ -458,3 +462,44 @@ The complexity becomes even worse when the elements are in reverse order. If ins
 - Can’t apply it to all data types since a suitable bucketing technique is required. Bucket sort’s efficiency is dependent on the distribution of the input values, thus it’s not worth it if your data are closely grouped.In many situations, you might achieve greater performance by using a specialized sorting algorithm like radix sort, counting sort, or burst sort instead of bucket sort.
 
 - Bucket sort’s performance is determined by the number of buckets used, which may need some additional performance adjustment when compared to other algorithms.
+
+    
+## DNF Sort
+
+DNF Sort (Dutch National Flag Sorting) , This is the sorting method which is specially designed for only the array which contain numbers `0's` `1's` and `2's` only.
+
```
Comment (Lakhankumawat) [inline S-SortingAlgorithms/README.md:470]: Add a working image of sorting algorithm small gif or any image which shows the process . Follow guidelines for README .

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

File: S-SortingAlgorithms/README.md (hunk 1 above)
Comment (Pranshu321) [reply S-SortingAlgorithms/README.md:473]: @Lakhankumawat  Working Image added.

**Review by Lakhankumawat — APPROVED** [review]
LGTM ✨🎉

**Pull request merged**